### ISSUE TITLE
Milestone 1: EDA - a literate Rmd of ipynb file

### ISSUE THREAD

**Issue by annychih** [UBC-MDS/DSCI_522_Group304#7]
Need to make either an .Rmd or .ipynb file and point to the file in the README.md file within the root of the public GitHub.com project repo

From Milestone 1:
Choose a format to submit your EDA, either an Rmd file or a Jupyter notebook. This should serve as a pre-cursor to the scripts you will eventually create. Feel free to experiment with different analyses and visualizations in this document. Data scientists often use a sandbox to play around in, before final scripts are created for automation. Similarly, it is valuable to try out a few different plots before finally settling on one or two for the final report. To get started, follow the EDA checklist in chapter 4 of the Art of Data Science.

Note - the EDA should be done in a literate code document (e.g., .Rmd or .ipynb file). If you choose .Rmd, also please include a rendered document that we can easily read on GitHub (e.g., .md or .pdf). These documents should live in a subdirectory, such as src or make one called eda. You should point to this file in the README of your public GitHub.com repo.

**Comment by annychih** [comment]
@zouwenjiao , nice clean code! We'll need to do a few things before the data is ready for EDA:

1. Delete the columns we don't need from the 2 datasets (keep columns A-L and the last 'score' column, but drop the rest since they don't match up between the 2 files)

2. Stack the files together so that the dataset is complete

3. Create 'subpopulation' datasets for our research questions (ex. 'All Students' 'Aboriginal and Non-Aboriginal') 

4. Could you please also drop the rows with 'Msk' for score? These are the schools/subgroups with not enough data.

**Comment by zouwenjiao** [comment]
1. I build up a frame for the EDA with everything in the checklist.

2. I added 
 - dataset description
 - load the data
 - explore the data
 - initial thoughts
 - research questions
 - data analysis and visualizations
 - summary and conclusion

3. In the wrangling part, I have delete some columns, drop the 'Msk's and stack the tables together

4. In the analysis part, I create 6 tables of summary statistics:
  - Public vs Independent
     - Mean and confidence interval of Numeracy scores in different subpopulations
     - Mean and confidence interval of Reading scores in different subpopulations
     - Mean and confidence interval of Writing scores in different subpopulations

 - Aboriginal vs Non-aboriginal
     - Mean and confidence interval of Numeracy scores
     - Mean and confidence interval of Reading scores 
     - Mean and confidence interval of Writing scores
@robilizando, you can use these 6 data frames to create viz.

**Comment by robilizando** [comment]
Added Preliminary plots and created a pull request. Sorry I could not do it in the morning, had merge-conflict problems that Tiffany helped me resolve.

**Comment by annychih** [comment]
I changed a few things this morning and added the following notes to the pull request:

1. Updated Robert’s chart titles to include a bit more detail about the dates (ex. 2017/18 – 2018/19).

2. Code Chunk 7.2.2: I changed the title and code chunk label to reflect what’s being shown and commented out the pub_ind_numeracy at the top of the cell because the plot doesn’t use just that data subset. I also changed the dataset to be filtered so it doesn’t use duplicated data (ie. Only one data_level and taking out summary ‘PROVINCE – TOTAL’ public_or_independent rows).

3. I couldn’t find data frames that were only ‘all aboriginal and non aboriginal’ (needed to make boxplots), so I made a sub_data group in the 5.2 code chunk

DATA_LEVEL was missing from the data, but it’s needed to make sure that school’s aren’t counted multiple times (ex. both in the School and Province data_level). I added the column to the main dataframe and went through to add the DATA_LEVEL == ‘SCHOOL LEVEL’ filter throughout (hopefully I didn’t miss any!).

This brings up a different issue though… if we simply average all the averages, we’re not considering the number of students each average applies to when we’re comparing groups (for example, we’re comparing mean scores for all independent schools vs public schools, but what if there are 1000x more public school students than independent school students? Each school is weighted evenly, when maybe they shouldn’t be).
